geopandas/_transformer.py

```diff
@@ -0,0 +1,13 @@
+from pyproj import Transformer as PyprojTransformer
+from pyproj import __version__ as pyproj_version
+
+# Setup cached Transformer for pyproj 2.x and 3.x
+if pyproj_version.startswith(("2", "3")):
```

Comment:
We only support pyproj >= 2.6 (and soon maybe just >= 3), so this if/else is not needed. Given that, I think it's probably not worth putting this in a separate file, and I would just inline the `lru_cache` in the array.py file.